src/systemctl/systemctl-start-unit.c

```diff
@@ -348,7 +348,18 @@ int verb_start(int argc, char *argv[], void *userdata) {
         }
 
         if (arg_wait) {
-                r = bus_call_method_async(bus, NULL, bus_systemd_mgr, "Subscribe", NULL, NULL, NULL);
+                bool new_signal_available = false;
+
+                r = bus_subscribe_with_flags_cached(bus);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to determine whether JobRemovedEx signal is available, falling back to legacy JobRemoved: %m");
+                else
+                        new_signal_available = r > 0;
+
+                if (new_signal_available)
+                        r = bus_call_method_async(bus, NULL, bus_systemd_mgr, "SubscribeWithFlags", NULL, NULL, "t", 0);
+                else
+                        r = bus_call_method_async(bus, NULL, bus_systemd_mgr, "Subscribe", NULL, NULL, NULL);
```

Comment:
If PID1 has `SubscribeWithFlags` method, then the method is called twice.

Maybe, `bus_subscribe_with_flags_cached()` always call `SubscribeWithFlags` or `Subscribe`.
And, also make them called with async.

Reply:
It cannot be async, as we need to wait for the result before we know which one to call. It won't hurt to queue the second async call, it will just be a no-op.

Reply:
I mean, this is still incomplete, but I think we should do something like that: https://github.com/yuwata/systemd/commit/5fcb45500f00e2d35a22a4a557bbb0473322fb67.

Currently, all `Subscribe` method calls and `JobRemoved` signal match installations are done in asynchronously. I think we should use `bus_wait_for_jobs_new()` everywhere, and it handles Ex and non-Ex gracefully.